### PR TITLE
coff files are typically called .obj

### PR DESCRIPTION
**Pull request by abadams**

### CODE REVIEW

File: HalideGenerator.cmake [hunk 1]
```diff
@@ -77,10 +77,10 @@ function(halide_add_generator_dependency)
   # generated library
   if (WIN32)
```
Comment (dsharletg) [inline HalideGenerator.cmake]: Is WIN32 a flag for windows, or MSVC specifically? .o still makes sense on windows, if the cmake generator is mingw (or something else).

File: src/Output.cpp [hunk 2]
```diff
@@ -30,8 +36,16 @@ void compile_module_to_assembly(const Module &module, std::string filename)  {
...
-    if (assembly_filename.empty()) assembly_filename = module.name() + ".s";
+    if (object_filename.empty()) {
+        if (module.target().os == Target::Windows) {
+            object_filename = module.name() + ".obj";
+        } else {
+            object_filename = module.name() + ".o";
+        }
+    }
+    if (assembly_filename.empty()) {
+        assembly_filename = module.name() + ".s";
```
Comment (zvookin) [inline src/Output.cpp]: Windows convention is typically ".asm", though perhaps that depends wich assembler one uses. (I've seen ".nasm" as well.) Perhaps the .o vs. .obj distinction is for something other than visual convention however and .s can be left as is.

**Comment by zvookin** [comment]
LGTM...

File: HalideGenerator.cmake (hunk 1 above)
Comment (abadams) [inline HalideGenerator.cmake]: It's something that you're about to run lib.exe on, so I think msvc conventions are probably the right thing.

**Comment by delcypher** [comment]
@abadams This is going to break #855. It currently doesn't use ``halide_add_generator_dependency()`` and thus reimplements some of its logic. The reason for doing this is because

* Some of the generators do not understand the command line arguments that will be passed and will crash if they are passed.
* ``halide_add_generator_dependency()`` does not understand compiling the Halide runtime as a separate object file.

**Comment by abadams** [comment]
Is there some equivalent change that would make #855 use .obj as a suffix
when on windows?

**Comment by abadams** [comment]
Actually I'm confused - I see a bunch of changes to the apps in #855, but most of those are not generators, and HalideGenerator.cmake appears unchanged. Are you just saying that those .o" literals in the app CMake files need to be .obj if it's on Windows? So we just need to set a cmake variable somewhere to the object file suffix, if there isn't one already.

**Comment by delcypher** [comment]
@abadams I mostly only fixed apps that were already list in the [CMakelists.txt file](https://github.com/delcypher/Halide/blob/3bcb22b6439f271561742e589db6171732484450/apps/CMakeLists.txt). All of the apps I fixed do AOT (apart from Interpolate)

Sorry when I say "generators" I mean they do AOT, I don't mean they use the ``Halide::Generator`` class.

Changing the ``CMakeLists.txt`` files to use ``CMAKE_CXX_SOURCE_FILE_EXTENSIONS`` instead of hardcoding ``.o`` would work. I brought this up because I assumed ``halide_add_generator_dependency()`` was intended to work with AOT in general and not specifically the ``Halide::Generator`` class. Is that assumption wrong?

**Comment by delcypher** [comment]
@abadams When you merge this PR I'll rebase and fix my PR so please don't wait to merge this on my account.

**Comment by abadams** [comment]
Gotcha - "generator" is used interchangeably and I was assuming the other use. I merged #855 and now I'm testing this branch on windows. Some of the apps are hard-coded to produce a .o, so the changes to the CMakefiles from 855 are a little patchy.

**Comment by abadams** [comment]
... except the apps don't currently build under windows, so my test on windows trivially passed

**Comment by delcypher** [comment]
@abadams Sorry I forgot I did this. I was originally planning to have some of the apps working under windows but I hit some issues that I thought would be easier to address in another PR at some point in the future.

**Comment by abadams** [comment]
I don't think the apps ever built under windows - e.g. the camera pipe test includes arpa/inet.h

Do the cmakefile changes look sane to you?

**Comment by delcypher** [comment]
@abadams LGTM

**Pull request merged**